### PR TITLE
fix: #952 eliminate summary page

### PR DESCRIPTION
**Pull request by NickSaglioni**
- Removing Summary Page and all references to it including components;
- Removing grantAccessDataState because it was created just to be able to move back from summary page to grant access page keeping the form data for edition;
- Code clean up and optimizations;

### CODE REVIEW

File: frontend/src/components/grantaccess/GrantAccess.vue [hunk 1]
```diff
@@ -216,17 +209,88 @@ function areVerificationsPassed() {
...
+                `${
+                    formData.value.userId
+                } was not added with Client IDs: ${errorList.join(', ')}`
+            );
+        }
+
+        router.push('/dashboard');
+    } else {
+        const data = toRequestPayload(formData.value, null);
+        await userRoleAssignmentApi.createUserRoleAssignment(data);
```
Comment (MCatherine1994) [inline frontend/src/components/grantaccess/GrantAccess.vue:285]: maybe we should catch the error for this request as well? But I would suggest confirm with @OlgaLiber2 first, how we would like to handle the error case when the role already assigned to the user. For now, we handle them differently for adding a role with forest client number, and without. One uses warning message, one uses toast message. Maybe cut another ticket to confirm with Olga and make them consistent. 

Screen Shot 2023-10-24 at 3 24 45 PM
Screen Shot 2023-10-24 at 3 26 27 PM

**Comment by MCatherine1994** [comment]
Thanks Nick! Overall looks good to me. Just not sure if these buttons color are same as in the design, but maybe you can check and adjust all the style in the ticket #844
Screen Shot 2023-10-24 at 3 30 08 PM

**Comment by NickSaglioni** [comment]
> Thanks Nick! Overall looks good to me. Just not sure if these buttons color are same as in the design, but maybe you can check and adjust all the style in the ticket #844
> Screen Shot 2023-10-24 at 3 30 08 PM

Yes, I did not adjust the styling because it is going to be done in ticket #844. I just handled the buttons now because the text needed to be changed for this task, but the styling I was going to handle in ticket #844.

File: frontend/src/components/grantaccess/GrantAccess.vue (hunk 1 above)
Comment (NickSaglioni) [reply frontend/src/components/grantaccess/GrantAccess.vue:285]: I makes sense to be the same, I'll adjust it.

**Review by MCatherine1994 — APPROVED** [review]

**Comment by MCatherine1994** [comment]
Hi Nick, thanks! This looks good to me know, approved. Could you please create another ticket for this error case? When there is an error from backend, we display the error message in both notification and toast. 


Screen Shot 2023-10-25 at 11 21 37 AM

**Comment by NickSaglioni** [comment]
> Hi Nick, thanks! This looks good to me, approved. Could you please create another ticket for this error case? When there is an error from backend, we display the error message in both notification and toast.
> 
> Screen Shot 2023-10-25 at 11 21 37 AM

Thanks Catherine! I'll open a ticket for that.

**Comment by MCatherine1994** [comment]
> > Hi Nick, thanks! This looks good to me, approved. Could you please create another ticket for this error case? When there is an error from backend, we display the error message in both notification and toast.
> > Screen Shot 2023-10-25 at 11 21 37 AM
> 
> Thanks Catherine! I'll open a ticket for that.

Hi Nick, just a reminder, let's create this new ticket before the next sprint.

**Pull request merged**